Move CompiledContract and CompiledContractCache to near-vm-errors

Ekleog-NEAR opened the pull request:
Their actual location should be near-vm-runtime. However, near-vm-runtime currently depends on near-primitives, and near-primitives needs to have access to CompiledContract and CompiledContractCache. So it would not be easy to move them to near-vm-runtime right now.

However, the end-goal of this refactoring is actually to be able to merge near-vm-errors and near-vm-runtime. So even if we move them to the "wrong" place right now, it will become the "right" place as soon as the other refactorings necessary for that merge to happen will be complete.

Longarithm reviewed (changes requested):
> However, the end-goal of this refactoring is actually to be able to merge near-vm-errors and near-vm-runtime.

It makes sense, but having a struct there which is not an error still hurts the eye. What if we temporarily rename `near-vm-errors` with `near-vm-helpers`? Or add a TODO (global issue ID) above these structs to merge the crates you mentioned.

Ekleog-NEAR commented:
@Longarithm I’m hoping to get the merge done by EOW, so a rename would probably be superfluous noise. So I’ve opened #9180 to track merging the merge and referenced it from the source code, hopefully this addresses your comment :)
